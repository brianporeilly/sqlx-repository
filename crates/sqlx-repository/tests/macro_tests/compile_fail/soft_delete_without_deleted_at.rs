//! Test that soft delete requires deleted_at field

use sqlx_repository::prelude::*;
use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, sqlx::FromRow, Repository)]
#[repository(table = "users")]
#[repository(soft_delete)]  // This should fail without deleted_at field
pub struct User {
    pub id: i32,
    pub name: String,
    pub email: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    // Missing: pub deleted_at: Option<DateTime<Utc>>,
}

fn main() {}