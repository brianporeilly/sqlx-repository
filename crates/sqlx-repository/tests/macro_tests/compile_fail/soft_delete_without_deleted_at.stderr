error: Soft delete enabled but no 'deleted_at' field found. Add 'pub deleted_at: Option<DateTime<Utc>>' to your struct
  --> tests/macro_tests/compile_fail/soft_delete_without_deleted_at.rs:8:1
   |
8  | / #[repository(table = "users")]
9  | | #[repository(soft_delete)]  // This should fail without deleted_at field
10 | | pub struct User {
11 | |     pub id: i32,
...  |
17 | | }
   | |_^
